pmlc/dialect/op_lib/cpp/utils.h

```diff
@@ -0,0 +1,98 @@
+//===- utils.cc - MLIR op lib dialect tblgen utils for C++ ----------------===//
+//
+// Copyright 2019 Intel Corporation.
+//
+// =============================================================================
+//
+// Tablegen utilities for the C++ outputs of the op lib dialect. This
+// encompasses things such as headers, includes, and type mappings.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <map>
+
+#include "mlir/Support/STLExtras.h"
+#include "llvm/ADT/StringExtras.h"
+
+using mlir::StringRef;
+
+namespace pmlc::dialect::op::tblgen::cpp {
+
+static inline const char* const fileCommentHeader = R"( // Copyright 2019 Intel Corporation.
+/*===- TableGen'erated file -------------------------------------*- C++ -*-===*\
+|*                                                                            *|
+|* Op Lib C++ EDSL Wrapper                                                    *|
+|*                                                                            *|
+|* Automatically generated file, do not edit!                                 *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+)";
+
+static inline const char* const includeHeader = R"(
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "plaidml2/edsl/edsl.h"
+#include "plaidml2/op/ffi.h"
+
+)";
```

Comment:
Ya, I think we should not really do any of this stuff from the generated file, instead if the outside file `ops.cc` did it, then it would make it much easier to create and use test dialects.